src/Symfony/Bundle/FrameworkBundle/Debug/CliRequest.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Debug;
+
+use Symfony\Bundle\FrameworkBundle\Console\TraceableCommand;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * @internal
+ */
+class CliRequest extends Request
+{
+    public function __construct(public readonly TraceableCommand $command)
+    {
+        parent::__construct(
+            attributes: ['_controller' => \get_class($command->command), '_virtual' => 'command'],
+            server: $_SERVER,
+        );
+    }
+
+    // Methods below allow to populate a profile, thus enable search and filtering
+    public function getUri(): string
+    {
+        if ($this->server->has('SYMFONY_CLI_BINARY_NAME')) {
+            $binary = $this->server->get('SYMFONY_CLI_BINARY_NAME').' console';
+        } else {
+            $binary = $this->server->get('argv')[0];
+        }
+
+        return $binary.' '.$this->command->input;
+    }
+
+    public function getMethod(): string
+    {
+        return $this->command->isInteractive ? 'INTERACTIVE' : 'BATCH';
+    }
+
+    public function getResponse(): Response
+    {
+        return new class($this->command->exitCode) extends Response {
+            public function __construct(private readonly int $exitCode)
+            {
+                parent::__construct();
+            }
+
+            public function getStatusCode(): int
+            {
+                return $this->exitCode;
```

Comment:
This is very, very hacky 😄. Maybe should we at least:

- respect the HTTP semantics: return 200 for the 0 exit code, and 500 for other codes
- create a `CliResponse` class with a `getExitStatus()` method (or just add this method to the request class)

Reply:
The profiler uses the response to define the status code of a profile.
The profile is too much bound to HTTP and IMHO it does not worth adding more complexity to it for now.
We should maybe iterate on this, as part as what you said in https://github.com/symfony/symfony/pull/47416#issuecomment-1736139138.